crates/p2p/src/tests/common.rs

```diff
@@ -118,18 +186,31 @@ impl Setup {
         })
     }
 
-    /// Create `n` random keypairs, peer ids from them and sequential in-memory
+    /// Create `n` random keypairs, transport ids from them and sequential in-memory
     /// addresses.
     fn setup_keys_ids_addrs_of_n_users(
         n: usize,
-    ) -> (Vec<Keypair>, Vec<PeerId>, Vec<libp2p::Multiaddr>) {
-        let keypairs = (0..n)
+    ) -> (
+        Vec<Keypair>,
+        Vec<Keypair>,
+        Vec<PeerId>,
+        Vec<libp2p::Multiaddr>,
```

Comment:
This is a BIG tuple that you're returning.
Can we have a proper test `struct` to hold these for bette maintainability and understanding what's going on?